opal/util/json/opal_json.c

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (c) 2024      Amazon.com, Inc. or its affiliates.
+ *                         All Rights reserved.
+ * $COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * $HEADER$
+ */
+
+#include "opal/util/json/opal_json.h"
+#include "opal/constants.h"
+#include "opal/util/json/3rd-party/json.h"
+#include "opal/util/output.h"
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define CHECK_OBJ_TYPE(obj, expected)                                                \
+    do {                                                                             \
+        if ((obj)->type != (expected)) {                                             \
+            opal_output_verbose(1, 0, "%s only supports %s\n", __func__, #expected); \
+            return OPAL_ERROR;                                                       \
+        }                                                                            \
+    } while (0)
+
+struct opal_json_internal_t {
+    struct opal_json_t parent;
+    json_value *value;
+};
+typedef struct opal_json_internal_t opal_json_internal_t;
+
+static void opal_json_internal_free(opal_json_internal_t *json)
+{
+    if (NULL == json) {
+        return;
+    }
+
+    /* The root JSON object will release the memory of its children */
+    if (json->value && NULL == json->value->parent) {
+        json_value_free(json->value);
+    }
+
+    free(json);
+}
+
+static inline int opal_json_internal_translate_type(json_type type, opal_json_type *out)
+{
+    int ret = OPAL_SUCCESS;
+    switch (type) {
+    case json_null:
+        *out = OPAL_JSON_NULL;
+        break;
+    case json_array:
+        *out = OPAL_JSON_ARRAY;
+        break;
+    case json_boolean:
+        *out = OPAL_JSON_BOOL;
+        break;
+    case json_integer:
+        *out = OPAL_JSON_INTEGER;
+        break;
+    case json_double:
+        *out = OPAL_JSON_DOUBLE;
+        break;
+    case json_string:
+        *out = OPAL_JSON_STRING;
+        break;
+    case json_object:
+        *out = OPAL_JSON_OBJECT;
+        break;
+    default:
+        opal_output_verbose(1, 0, "Invalid JSON value type %d\n", type);
```

Comment:
Shouldn't this be a show_help, not a direct opal_output_verbose?  (same goes for all cases of errors emitted via opal_output_verbose)

Reply:
Thanks. I switched to show help.